tests/Mutator/Unwrap/UnwrapArrayReverseTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator\Unwrap;
+
+use Infection\Tests\Mutator\AbstractMutatorTestCase;
+
+/**
+ * @internal
+ */
+final class UnwrapArrayReverseTest extends AbstractMutatorTestCase
+{
+    /**
+     * @dataProvider provideMutationCases
+     */
+    public function test_mutator($input, $expected = null): void
+    {
+        $this->doTest($input, $expected);
+    }
+
+    public function provideMutationCases(): \Generator
```

Comment:
For the sake of completeness, sure why not.

Reply:
@sanmai 

Not sure about the number of tests, do we need to have three instead of one data set

* as is at the moment
* with `$preserve_keys` set to `false`
* with `$preserve_keys` set to `true`

or will two more test cases suffice?

Reply:
I'd leave it with just one test with an extra argument. It can be a variable like so:
```
- $a = array_reverse(['foo'], $preserve_keys);
+ $a = ['foo'];
```